pkg/connector/service.go

```diff
@@ -275,3 +284,27 @@ func (s *Service) SetState(ctx context.Context, id string, state any) (*Instance
 
 	return conn, err
 }
+func (s *Service) validateConnector(cfg Config, id string) error {
+	// contains all the errors occurred while provisioning configuration files.
+	var multierr error
+
+	if cfg.Name == "" {
+		multierr = multierror.Append(multierr, ErrNameMissing)
+	}
+	if len(cfg.Name) > 64 {
+		multierr = multierror.Append(multierr, ErrNameOverLimit)
+	}
+	if id == "" {
+		multierr = multierror.Append(multierr, ErrIDMissing)
+	} else {
```

Comment:
If we change the regex to allow any number of characters (`^[A-Za-z0-9-_:]*$`) we don't need this `else`, which would make the function a bit more readable, as all validations are independently executed for all strings. Same in the pipeline validation.